kornia/losses/perceptual.py

```diff
@@ -0,0 +1,324 @@
+from typing import Callable, Tuple
+
+import torch
+import torch.nn as nn
+import torchvision
+
+import kornia
+from kornia import testing
+from kornia.geometry.transform.imgwarp import get_perspective_transform, warp_perspective
+
+
+class LossNetwork(nn.Module):
+
+    def __init__(
+        self,
+        model_name: str = 'resnet34',
+        output_layer: int = 1,
+        freeze: bool = True,
+    ) -> None:
+        super().__init__()
+        if 'resnet' not in model_name:
+            raise ValueError('Only ResNets are supported ATM')
+
+        # Define resnet model
+        loss_network_fn = getattr(torchvision.models, model_name)
+        self.resnet = loss_network_fn(pretrained=True, progress=False)
+
+        # Clear unnecessary layers
+        self.output_layer = output_layer
+        if self.output_layer < 2:
+            self.resnet.layer2 = torch.nn.Identity()
+        if self.output_layer < 3:
+            self.resnet.layer3 = torch.nn.Identity()
+        if self.output_layer < 4:
+            self.resnet.layer4 = torch.nn.Identity()
+        self.resnet.avgpool = torch.nn.Identity()
+        self.resnet.fc = torch.nn.Identity()
+
+        # Freeze the model
+        if freeze:
+            for param in self.resnet.parameters():
+                param.requires_grad = False
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+
+        if x.shape[1] == 1:
+            x = x.repeat(1, 3, 1, 1)
+
+        x = self.resnet.conv1(x)
+        x = self.resnet.bn1(x)
+        x = self.resnet.relu(x)
+        x = self.resnet.maxpool(x)
+
+        x = self.resnet.layer1(x)
+
+        if self.output_layer > 1:
+            x = self.resnet.layer2(x)
+        if self.output_layer > 2:
+            x = self.resnet.layer3(x)
+        if self.output_layer > 3:
+            x = self.resnet.layer4(x)
+
+        return x
+
+
+def _image_shape_to_corners(image: torch.Tensor) -> torch.Tensor:
+    """Convert image size to 4 corners representation in clockwise order.
+
+    Args:
+        image: image tensor with shape :math:`(B, C, H, W)` where B = batch size,
+            C = number of channels
+
+    Return:
+        the corners of the image.
+    """
+    testing.KORNIA_CHECK_SHAPE(image, ['*', '*', '*', '*'])
```

Comment:
`from kornia.core.check import KORNIA_CHECK_SHAPE`
```suggestion
    KORNIA_CHECK_SHAPE(image, ['B', 'C', 'H', 'W'])
```